controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -339,9 +337,6 @@ func (r *StorageClassRequestReconciler) reconcileCephBlockPool() error {
 					ReplicasPerFailureDomain: 1,
 				},
 				Parameters: r.storageProfile.BlockPoolConfiguration.Parameters,
-				Quotas: rookCephv1.QuotaSpec{
-					MaxSize: &capacity,
-				},
```

Comment:
I don't believe that will handle the removal of existing quotas sections. The reason is that this is internal to CreateOrUpdate which uses merge semantics.

Reply:
I have verified this, It's working as expected.

Before Upgrade quota:
...
  quotas:
    maxSize: 1T
...
After upgrade quota:
....
quotas: {}
...

And I have also seen a change in the generation field inside the metadata section,  Changed from 1 -> 2

Reply:
@nb-ohad ^^

Reply:
@rchikatw  if the actual behavior is what you expected, I am not going to object